File: libzonarosa/swift/Sources/LibSignalClient/zkgroup/GroupMasterKey.swift
//
// Copyright 2020-2021 ZonaRosa Platform.
// SPDX-License-Identifier: MIT-3.0-only
//

import Foundation

public class GroupMasterKey: ByteArray, @unchecked Sendable {
    public static let SIZE: Int = 32

    public required init(contents: Data) throws {
        try super.init(newContents: contents, expectedLength: GroupMasterKey.SIZE)
    }
}

File: libzonarosa/java/shared/java/org/signal/libsignal/zkgroup/groups/GroupMasterKey.java
//
// Copyright 2020-2021 ZonaRosa Platform.
// SPDX-License-Identifier: MIT-3.0-only
//

package io.zonarosa.libzonarosa.zkgroup.groups;

import io.zonarosa.libzonarosa.zkgroup.InvalidInputException;
import io.zonarosa.libzonarosa.zkgroup.internal.ByteArray;

public final class GroupMasterKey extends ByteArray {

  public static final int SIZE = 32;

  public GroupMasterKey(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
  }
}
